Mathlib/Analysis/NormedSpace/Alternating/Basic.lean

```diff
@@ -0,0 +1,599 @@
+`‖f m‖` is bounded by `‖f‖` times the product of the `‖m i‖`, `nnnorm` version. -/
+theorem le_opNNNorm (f : E [⋀^ι]→L[𝕜] F) (m : ι → E) : ‖f m‖₊ ≤ ‖f‖₊ * ∏ i, ‖m i‖₊ :=
+  f.1.le_opNNNorm m
+
+theorem le_of_opNNNorm_le {C : ℝ≥0} (h : ‖f‖₊ ≤ C) (m : ι → E) : ‖f m‖₊ ≤ C * ∏ i, ‖m i‖₊ :=
+  f.1.le_of_opNNNorm_le h m
+
+theorem opNNNorm_le_iff {C : ℝ≥0} : ‖f‖₊ ≤ C ↔ ∀ m, ‖f m‖₊ ≤ C * ∏ i, ‖m i‖₊ :=
+  f.1.opNNNorm_le_iff
+
+theorem isLeast_opNNNorm (f : E [⋀^ι]→L[𝕜] F) :
+    IsLeast {C : ℝ≥0 | ∀ m, ‖f m‖₊ ≤ C * ∏ i, ‖m i‖₊} ‖f‖₊ :=
+  f.1.isLeast_opNNNorm
+
+theorem opNNNorm_prod (f : E [⋀^ι]→L[𝕜] F) (g : E [⋀^ι]→L[𝕜] G) :
+    ‖f.prod g‖₊ = max (‖f‖₊) (‖g‖₊) :=
+  f.1.opNNNorm_prod g.1
+
+theorem opNorm_prod (f : E [⋀^ι]→L[𝕜] F) (g : E [⋀^ι]→L[𝕜] G) : ‖f.prod g‖ = max (‖f‖) (‖g‖) :=
+  f.1.opNorm_prod g.1
+
+theorem opNNNorm_pi {ι' : Type*} [Fintype ι'] {F : ι' → Type*} [∀ i', SeminormedAddCommGroup (F i')]
+    [∀ i', NormedSpace 𝕜 (F i')] (f : ∀ i', E [⋀^ι]→L[𝕜] F i') : ‖pi f‖₊ = ‖f‖₊ :=
+  ContinuousMultilinearMap.opNNNorm_pi fun i ↦ (f i).1
+
+theorem opNorm_pi {ι' : Type*} [Fintype ι'] {F : ι' → Type*} [∀ i', SeminormedAddCommGroup (F i')]
+    [∀ i', NormedSpace 𝕜 (F i')] (f : ∀ i', E [⋀^ι]→L[𝕜] F i') : ‖pi f‖ = ‖f‖ :=
+  ContinuousMultilinearMap.opNorm_pi fun i ↦ (f i).1
+
+instance instNormedSpace {𝕜' : Type*} [NormedField 𝕜'] [NormedSpace 𝕜' F] [SMulCommClass 𝕜 𝕜' F] :
+    NormedSpace 𝕜' (E [⋀^ι]→L[𝕜] F) :=
+  ⟨fun c f ↦ f.1.opNorm_smul_le c⟩
+
+section
+
+@[simp] theorem norm_ofSubsingleton [Subsingleton ι] (i : ι) (f : E →L[𝕜] F) :
+    ‖ofSubsingleton 𝕜 E F i f‖ = ‖f‖ :=
+  ContinuousMultilinearMap.norm_ofSubsingleton i f
+
+@[simp] theorem nnnorm_ofSubsingleton [Subsingleton ι] (i : ι) (f : E →L[𝕜] F) :
+    ‖ofSubsingleton 𝕜 E F i f‖₊ = ‖f‖₊ :=
+  NNReal.eq <| norm_ofSubsingleton i f
+
+/-- `ContinuousAlternatingMap.ofSubsingleton` as a linear isometry. -/
+@[simps (config := {simpRhs := true})]
+def ofSubsingletonLIE [Subsingleton ι] (i : ι) : (E →L[𝕜] F) ≃ₗᵢ[𝕜] (E [⋀^ι]→L[𝕜] F) where
+  __ := ofSubsingleton 𝕜 E F i
+  map_add' _ _ := rfl
+  map_smul' _ _ := rfl
+  norm_map' := norm_ofSubsingleton i
+
+theorem norm_ofSubsingleton_id_le [Subsingleton ι] (i : ι) :
+    ‖ofSubsingleton 𝕜 E E i (.id _ _)‖ ≤ 1 :=
+  ContinuousMultilinearMap.norm_ofSubsingleton_id_le ..
+
+theorem nnnorm_ofSubsingleton_id_le [Subsingleton ι] (i : ι) :
+    ‖ofSubsingleton 𝕜 E E i (.id _ _)‖₊ ≤ 1 :=
+  ContinuousMultilinearMap.nnnorm_ofSubsingleton_id_le ..
+
+variable (𝕜 E)
+
+@[simp] theorem norm_constOfIsEmpty [IsEmpty ι] (x : F) : ‖constOfIsEmpty 𝕜 E ι x‖ = ‖x‖ :=
+  ContinuousMultilinearMap.norm_constOfIsEmpty _ _ _
+
+@[simp] theorem nnnorm_constOfIsEmpty [IsEmpty ι] (x : F) : ‖constOfIsEmpty 𝕜 E ι x‖₊ = ‖x‖₊ :=
+  NNReal.eq <| norm_constOfIsEmpty _ _ _
+
+end
+
+variable (𝕜 E F G) in
+/-- `ContinuousAlternatingMap.prod` as a `LinearIsometryEquiv`. -/
+@[simps]
+def prodₗᵢ : (E [⋀^ι]→L[𝕜] F) × (E [⋀^ι]→L[𝕜] G) ≃ₗᵢ[𝕜] (E [⋀^ι]→L[𝕜] (F × G)) where
```

Comment:
Naming question: this one has li as a subscript, while `ofSubsingletonLIE` has a different convention. Is it possible to settle on one convention and be coherent about it?

Reply:
Changed to `LI`/`LIE` so that we don't rely on the notation too much.